docs/core/deploying/trim-self-contained.md

```diff
@@ -7,14 +7,15 @@ ms.date: 04/03/2020
 ---
 # Trim self-contained deployments and executables
 
-When publishing an application self-contained, the .NET Core runtime is bundled together with the application. This bundling adds a significant amount of content to your packaged application. When it comes to deploying your application, size is often an important factor. Keeping the size of the package application as small as possible is typically a goal for application developers.
+The globally installed runtime and framework libraries has been the most successful runtime deployment model since the inception of .NET. In this scenario, the application developer will only bundle the application and third-party assemblies with the expectation that the .NET runtime and framework will be available in the client machine. This deployment model will continue to be the dominant one in .NET Core as well but there will be other application deployment models targeted at app developers where the above model is not the best one. One such one is targeted for some application developers that want to publish a [self-contained application](runtime-patch-selection.md) where the .NET Core runtime and framework are bundled together with the application.
 
-Depending on the complexity of the application, only a subset of the runtime is required to run the application. These unused parts of the runtime are unnecessary and can be trimmed from the packaged application.
+Trim self-contained deployment model is a specialized version of self-contained deployment model and is optimized for size that is a critical requirement for some client-side scenarios like Blazor applications. Depending on the complexity of the application, only a subset of the framework assemblies are required to run the application. These unused parts of the library are unnecessary and can be trimmed from the packaged application. However, there is a risk that the build time analysis of the application can cause failures at runtime, due to not being able to reliably analyze various problematic code patterns (largely centered on reflection use). So this deployment model will be offered as a preview feature where the build time analysis engine will provide warnings to the developer of code patterns that are problematic with the expectation that these code patterns will be fixed. Where possible, it is encouraged to move any runtime reflection dependencies in your application to build time by using code that meets the same requirements.
```

Comment:
Rephrasing, review for accuracy. It isn't clear, however, what "the same requirements" are that are mentioned at the end.

```suggestion
The trim-self-contained deployment model is a specialized version of the self-contained deployment model that is optimized to reduce deployment size. Minimizing deployment size is a critical requirement for some client-side scenarios like Blazor applications. Depending on the complexity of the application, only a subset of the framework assemblies are required to run the application. These unused parts of the library are unnecessary and can be trimmed from the packaged application. However, there is a risk that the build time analysis of the application can cause failures at runtime, due to not being able to reliably analyze various problematic code patterns (largely centered on reflection use). Because reliability can't be guaranteed, this deployment model is offered as a preview feature. The build time analysis engine provides warnings to the developer of code patterns that are problematic, with the expectation that these code patterns will be fixed. Where possible, we recommend that you move any runtime reflection dependencies in your application to build time by using code that meets the same requirements.
```